File: ogani-frontend/src/services/category.ts
import api from "./api"
import type { IProduct } from "./product";

export interface ICategory {
  id: number;
  name: string;
  image_url: string;
  Products?: IProduct[];
}

export const getCategories = () => {
  return api.get("/categories");
};

export const getCategoryDetail = ( id: number ) => {
  return api.get<{ category: ICategory }>(`/categories/${id}`);
};

File: ogani-backend/models/category.go
package models

import (
	"time"
)

type Category struct {
	ID        uint       `json:"id" gorm:"primaryKey"`
	Name      string     `json:"name" gorm:"unique"`
	ImageUrl  string     `json:"image_url"`
	CreatedAt *time.Time `json:"created_at"`
	UpdatedAt *time.Time `json:"updated_at"`

	Products []Product `gorm:"foreignKey:CategoryID"`
}
